plugins/aws/resoto_plugin_aws/resource/ecs.py

```diff
@@ -0,0 +1,1662 @@
+        "service_enable_execute_command": S("enableExecuteCommand"),
+    }
+    arn: Optional[str] = field(default=None)
+    status: Optional[str] = field(default=None)
+    cluster_arn: Optional[str] = field(default=None)
+    service_load_balancers: List[AwsEcsLoadBalancer] = field(factory=list)
+    service_registries: List[AwsEcsServiceRegistry] = field(factory=list)
+    service_desired_count: Optional[int] = field(default=None)
+    service_running_count: Optional[int] = field(default=None)
+    service_pending_count: Optional[int] = field(default=None)
+    service_launch_type: Optional[str] = field(default=None)
+    service_capacity_provider_strategy: List[AwsEcsCapacityProviderStrategyItem] = field(factory=list)
+    service_platform_version: Optional[str] = field(default=None)
+    service_platform_family: Optional[str] = field(default=None)
+    service_task_definition: Optional[str] = field(default=None)
+    service_deployment_configuration: Optional[AwsEcsDeploymentConfiguration] = field(default=None)
+    service_task_sets: List[AwsEcsTaskSet] = field(factory=list)
+    service_deployments: List[AwsEcsDeployment] = field(factory=list)
+    service_role_arn: Optional[str] = field(default=None)
+    service_events: List[AwsEcsServiceEvent] = field(factory=list)
+    service_placement_constraints: List[AwsEcsPlacementConstraint] = field(factory=list)
+    service_placement_strategy: List[AwsEcsPlacementStrategy] = field(factory=list)
+    service_network_configuration: Optional[AwsEcsNetworkConfiguration] = field(default=None)
+    service_health_check_grace_period_seconds: Optional[int] = field(default=None)
+    service_scheduling_strategy: Optional[str] = field(default=None)
+    service_deployment_controller: Optional[str] = field(default=None)
+    service_created_by: Optional[str] = field(default=None)
+    service_enable_ecs_managed_tags: Optional[bool] = field(default=None)
+    service_propagate_tags: Optional[str] = field(default=None)
+    service_enable_execute_command: Optional[bool] = field(default=None)
+
+    def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
+        # TODO add edge to Cloud Map service registry when applicable
+        if self.service_load_balancers:
+            for lb in self.service_load_balancers:
+                if lb.target_group_arn:
+                    builder.dependant_node(
+                        self,
+                        clazz=AwsAlbTargetGroup,
+                        arn=lb.target_group_arn,
+                    )
+                if lb.load_balancer_name:
+                    builder.dependant_node(
+                        self,
+                        clazz=AwsElb,
+                        name=lb.load_balancer_name,
+                    )
+
+        if self.service_task_definition:
+            task_def = self.service_task_definition
+            # task_def is either full arn OR "family:revision"
```

Comment:
Can we do the if/else here, instead of searching all nodes for nothing for at least one time?